### PR TITLE
Ensure basic DNS functionality before doing real work in configure-vm.sh

### LINKED ISSUE #8196: Node coming up as master on G[C|K]E
```
$ kubectl get nodes
NAME                                                  LABELS    STATUS
k8s-staging-regtest-loop-2015-05-13-10-29-51-node-1   <none>    Ready
k8s-staging-regtest-loop-2015-05-13-10-29-51-node-2   <none>    NotReady
k8s-staging-regtest-loop-2015-05-13-10-29-51-node-3   <none>    Ready
```

Node 2's kubelet logs:
```
E0513 17:35:31.557301   10906 kubelet.go:1298] error getting node: node k8s-staging-regtest-loop-2015-05-13-10-29-51-node-2 not found
E0513 17:35:31.666994   10906 kubelet.go:1767] Cannot get host IP: cannot get node: node k8s-staging-regtest-loop-2015-05-13-10-29-51-node-2 not found
I0513 17:35:31.667033   10906 manager.go:1222] Need to restart pod infra container for "etcd-server-k8s-staging-regtest-loop-2015-05-13-10-29-51-node-2_default" because it is not found
E0513 17:35:31.667968   10906 kubelet.go:1767] Cannot get host IP: cannot get node: node k8s-staging-regtest-loop-2015-05-13-10-29-51-node-2 not found
```

controller-manager side:
```
I0513 17:32:42.404862       7 nodecontroller.go:527] node k8s-staging-regtest-loop-2015-05-13-10-29-51-node-2 is never updated by kubelet
I0513 17:32:47.422920       7 nodecontroller.go:537] node k8s-staging-regtest-loop-2015-05-13-10-29-51-node-2 hasn't been updated for 1m5.422893194s. Last ready condition is: {Type:Ready Status:Unknown LastHeartbeatTime:2015-05-13 17:31:42 +0000 UTC LastTransitionTime:2015-05-13 17:32:42 +0000 UTC Reason:Kubelet never posted node status. Message:}
I0513 17:32:52.474976       7 nodecontroller.go:537] node k8s-staging-regtest-loop-2015-05-13-10-29-51-node-2 hasn't been updated for 1m10.474968033s. Last ready condition is: {Type:Ready Status:Unknown LastHeartbeatTime:2015-05-13 17:31:42 +0000 UTC LastTransitionTime:2015-05-13 17:32:42 +0000 UTC Reason:Kubelet never posted node status. Message:}
I0513 17:32:57.492246       7 nodecontroller.go:537] node k8s-staging-regtest-loop-2015-05-13-10-29-51-node-2 hasn't been updated for 1m15.492238494s. Last ready condition is: {Type:Ready Status:Unknown LastHeartbeatTime:2015-05-13 17:31:42 +0000 UTC L
```

I have a cluster to hand off if someone wants to debug. It's not a transient state: node 2 just failed to ever post a valid status.

cc @dchen1107 @davidopp 
cc @mbforbes @cjcullen (who looked at it briefly with me)

### PR DESCRIPTION
**Pull request by zmerlynn**
Fixes #8196. Maybe. If my theory is correct on how we got there. Also
changes the inference of master to be based on the master name, not
the node instance prefix. That way if we somehow have a bogus
hostname, the master will configure itself as a node, the whole
cluster fails, and it's a ton more obvious.

### CODE REVIEW

**Comment by zmerlynn** [comment]
Ping? This needs an assignment.

File: cluster/gce/configure-vm.sh [hunk 1]
```diff
@@ -24,6 +24,26 @@ is_push=$@
 readonly KNOWN_TOKENS_FILE="/srv/salt-overlay/salt/kube-apiserver/known_tokens.csv"
 readonly BASIC_AUTH_FILE="/srv/salt-overlay/salt/kube-apiserver/basic_auth.csv"
 
+function ensure-basic-networking() {
+  # Deal with GCE networking bring-up race. (We rely on DNS for a lot,
+  # and it's just not worth doing a whole lot of startup work if this
+  # isn't ready yet.)
+  until getent hosts metadata.google.internal &>/dev/null; do
```
Comment (bprashanth) [inline cluster/gce/configure-vm.sh]: For my own understanding, how long do we expect this to take/does it usually take?

**Comment by bprashanth** [comment]
LGTM

File: cluster/gce/configure-vm.sh (hunk 1 above)
Comment (zmerlynn) [inline cluster/gce/configure-vm.sh]: This was instantaneous when I tested it. I'm not actually entirely sure of my theory, but it's about the only thing that would explain what happened in that bug.

**Pull request merged**